src/runtime/src/core/map.rs

```diff
@@ -163,13 +182,32 @@ pub fn make_module() -> ValueMap {
                 .sort_by(|key_a, value_a, key_b, value_b| {
                     let value_a = match cache.get(key_a) {
                         Some(value) => value.clone(),
-                        None => get_sort_key(&mut cache, key_a, value_a),
+                        None => match get_sort_key(vm, &mut cache, key_a, value_a) {
```

Comment:
I think it would be good to avoid any further `vm` calls once an error has taken place, so maybe something like

```rust
if error.is_some() {
    return Ordering::Equal;
}
```
at the start of the `sort_by` function?

Reply:
A good point!